src/golang/lib/engine/authenticate.go

```diff
@@ -46,10 +50,16 @@ func AuthenticateLambdaConfig(ctx context.Context, authConf auth.Config) error {
 	}
 
 	for _, functionType := range functionsToShip {
-		err := lambda_utils.CreateLambdaFunction(functionType, lambdaConf.RoleArn)
-		if err != nil {
-			return errors.Wrap(err, "Unable to Create Lambda Function")
-		}
+		lambdaFunctionType := functionType
+		errGroup.Go(func() error {
+			return lambda_utils.CreateLambdaFunction(lambdaFunctionType, lambdaConf.RoleArn)
+		})
+		time.Sleep(3 * time.Second)
```

Comment:
This is interesting. I have no problem with a small sleep but I wonder what the concurrency issue is. If you do this a bunch of times, does a pattern emerge (eg. failures when pulling from ECR)? I have a hard time believing its random.